src/explorer/mavenExplorerProvider.ts

```diff
@@ -42,6 +42,21 @@ class MavenExplorerProvider implements TreeDataProvider<ITreeItem> {
         }
     }
 
+    public async addWorkspaceFolder(folder: vscode.WorkspaceFolder): Promise<void> {
+        await this.loadProjects(folder);
+        this.refresh();
+    }
+
+    public async removeWorkspaceFolder(folder: vscode.WorkspaceFolder): Promise<void> {
+        const pomPaths: string[] = await Utils.getAllPomPaths(folder);
+        for (const pomPath of pomPaths) {
+            if (this._projectMap.has(pomPath)) {
```

Comment:
Just delete it without checking the existence?

Since so far there is nothing else we want to do when the `pomPath` exists.

Reply:
I wrote in this way for clarification. "delete an un-existing key" looks strange. Anyway I'm ok with both, as they actually have no differences at all.